http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -97,6 +97,21 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         s"Begin app creation for app ${params.appName.value} in cloud context ${params.cloudContext.asString}"
       )
 
+      referenceDatabaseNames = app.appType.databases.collect { case ReferenceDatabase(name) => name }.toSet
+
+      // build WSM resource client
+      wsmResourceApi <- buildWsmResourceApiClient
+      wsmResourceDatabases <- F.delay(
+        wsmResourceApi
+          .enumerateResources(params.workspaceId.value, 0, 100, ResourceType.AZURE_DATABASE, StewardshipType.CONTROLLED)
+          .getResources
+          .asScala
+          .toList
+      )
+      referenceResourceDatabaseNames = wsmResourceDatabases
+        .filter(r => referenceDatabaseNames.contains(r.getMetadata().getName()))
+        .map(r => r.getResourceAttributes().getAzureDatabase().getDatabaseName())
```

Comment:
you'll see this logic repeated here and in `updateAndPollApp`. I intentionally did not pull this out into a function because it feels more readable this way, but I'm open to other opinions

Reply:
See above comment -- after looking at the code, I think it would read a bit cleaner as a separate method.